sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -319,6 +323,7 @@ class ResolveSessionCatalog(
     // session catalog and the table provider is not v2.
     case c @ ReplaceTableStatement(
          SessionCatalogAndTable(catalog, tbl), _, _, _, _, _, _, _, _, _) =>
+      assertNoNullTypeInSchema(c.tableSchema)
```

Comment:
It seems that we have a check for `ReplaceTableStatement`, but not for `ReplaceTableAsSelectStatement`. Is it okay?

Reply:
Yes. `case CreateTableAsSelectStatement` is not cover either.

Reply:
I will check the new REPLACE syntax with UT.

Reply:
By my testing, `ReplaceTableAsSelectStatement` should also check the assertion. I will change it.

Reply:
done